openc3/lib/openc3/microservices/microservice.rb

```diff
@@ -213,7 +213,7 @@ def shutdown
       @cancel_thread = true
       @microservice_status_sleeper.cancel if @microservice_status_sleeper
       MicroserviceStatusModel.set(as_json(:allow_nan => true), scope: @scope)
-      FileUtils.remove_entry(@temp_dir) if File.exist?(@temp_dir)
+      FileUtils.remove_entry_secure(@temp_dir, true)
```

Comment:
See https://docs.ruby-lang.org/en/3.2/FileUtils.html#method-c-remove_entry_secure. I think we were running into the TOCTTOU Vulnerability (https://docs.ruby-lang.org/en/3.2/FileUtils.html#module-FileUtils-label-Avoiding+the+TOCTTOU+Vulnerability) because we're explicitly using the `/tmp` directory everywhere.